tests/test_security_curve.cpp

```diff
@@ -229,12 +237,28 @@ int main (void)
     close_zero_linger (client);
 
 #ifdef ZMQ_BUILD_DRAFT_API
-    event = get_monitor_event (server_mon, NULL, NULL);
+    event = get_monitor_event (server_mon, NULL, NULL, 0);
+    assert (event == ZMQ_EVENT_HANDSHAKE_FAILED_ENCRYPTION);
+    // Two times because expect_bounce_fail involves two exchanges
+    event = get_monitor_event (server_mon, NULL, NULL, 0);
     assert (event == ZMQ_EVENT_HANDSHAKE_FAILED_ENCRYPTION);
+
+    // This event has to be the last one
+    event = get_monitor_event(server_mon, NULL, NULL, ZMQ_DONTWAIT);
+    assert (event == -1);
 #endif
 
     //  Check CURVE security with a garbage client public key
     //  This will be caught by the curve_server class, not passed to ZAP
+    
+    // This enables to flush the incoming monitoring events that disturbs the test.
+    // Even though the client socket is closed, the server still handles HELLO
+    // messages.
+    Sleep(250);
+    do {
+        event = get_monitor_event(server_mon, NULL, NULL, ZMQ_DONTWAIT);
+    } while(event != -1);
+
```

Comment:
@bluca I had to add this "dirty patch" because the server handles a HELLO message just before the test below.
I believe only the client would reconnect to the server, since the client is supposed to be closed I can't explain this behavior.